### DISCUSSION TITLE
Contributing: Running the Test Suite

### DISCUSSION THREAD

**Discussion by andrew8088** [SilentVoid13/Templater#212] [Help] [answered]
I'd love to contribute to Templater, both with internal variables/functions and functionality (I've actually fixed the dynamic templating bug #188 locally). However, I can't seem to find how to run the test suite.

- when I run `npm test`, it just generates the `main.test.js` file, but doesn't actually run the file
- when I run `node main.test.js`, I get `Cannot find module 'obsidian'`
- Based on the `Makefile`, there appears to be docker involved somehow, but there's no `Dockerfile`.

How can I run the tests?

**Comment by SilentVoid13** [comment, marked as answer]
Hey @andrew8088 thanks for considering contributing to Templater!
Since we can only access the obsidian code from the obsidian app, the `main.test.js` is actually an obsidian plugin!
You just have to add this file to a folder inside your vault `plugins` folder, and then run `Alt+t` (default hotkey) to launch the tests.
Tests results will appear in the console (`Ctrl+Shift+I` to open it)
Tests are very basic for now, I need to add some more. If you feel like it, don't hesitate to add some too. I'm using the `chai` assertion library. Just look at the existing tests and follow the structure, it's pretty basic.
